src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,15 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, we should rename but we want to keep bw comp for now.
+            if (this == INLINE) {
+                return "dynamic";
```

Comment:
Honestly I prefer inline, but all the user-facing stuff has been called dynamic, so I'd probably stick with that.